jsonrpc/http.go

```diff
@@ -30,33 +30,53 @@ func NewHTTP(rpc *Server, log utils.SimpleLogger) *HTTP {
 	return h
 }
 
-// ServeHTTP processes an incoming HTTP request
-func (h *HTTP) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+type httpConn struct {
+	req *http.Request
+	rw  http.ResponseWriter
+}
+
+func accept(rw http.ResponseWriter, req *http.Request) *httpConn {
 	if req.Method == "GET" {
 		status := http.StatusNotFound
 		if req.URL.Path == "/" {
 			status = http.StatusOK
 		}
-		writer.WriteHeader(status)
-		return
+		rw.WriteHeader(status)
+		return nil
 	} else if req.Method != "POST" {
-		writer.WriteHeader(http.StatusMethodNotAllowed)
-		return
+		rw.WriteHeader(http.StatusMethodNotAllowed)
+		return nil
+	}
+
+	return &httpConn{
+		req: req,
+		rw:  rw,
+	}
+}
+
+func (c *httpConn) Read(p []byte) (int, error) {
+	c.req.Body = http.MaxBytesReader(c.rw, c.req.Body, MaxRequestBodySize)
```

Comment:
we need to move `http.MaxBytesReader` to `accept`, otherwise caller can read more than `MaxRequestBodySize` by reading in chunks.

I think httpConn can store an io.ReaderCloser instead of the entire `http.Request`